spring-data-relational/src/main/java/org/springframework/data/relational/core/mapping/DefaultAggregatePath.java

```diff
@@ -194,6 +209,47 @@ public AggregatePath getIdDefiningParentPath() {
 		return AggregatePathTraversal.getIdDefiningPath(this);
 	}
 
+	@Override
+	public AggregatePath getTail() {
+
+		if (getLength() <= 2) {
+			return null;
+		}
+
+		AggregatePath tail = null;
+		for (RelationalPersistentProperty prop : this.path) {
+			if (tail == null) {
+				tail = context.getAggregatePath(context.getPersistentEntity(prop));
+			} else {
+				tail = tail.append(prop);
+			}
+		}
+		return tail;
+	}
+
+	@Override
+	@Nullable
+	public AggregatePath substract(@Nullable AggregatePath basePath) {
```

Comment:
This method is suspicious. Going back in the model indicates that something else went wrong much earlier and we're trying to compensate for it here.

Reply:
It's used for `ColumnInfos` which has a `basePath` which is the common start of all the paths referenced. 
Let me know if you want that changed in any way.